scripts/simplify_network.py

```diff
@@ -522,6 +523,151 @@ def consense(x):
     return clustering.network, clustering.busmap
 
 
+def remove_empty_isolated_nodes(n):
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    n.determine_network_topology()
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+
+    # merge buses having sufficient attached load
+    loads_t_sum = n.loads_t.p_set.sum(axis=0)
+    load_islands = loads_t_sum.loc[loads_t_sum.index.isin(i_islands)]
+
+    # islands without generation and generation should be dropped
+    generation_islands = n.generators.loc[n.generators.bus.isin(i_islands)]
+    isl_no_load = set(i_islands) - set(load_islands.index)
+    isl_no_gen = set(i_islands) - set(generation_islands.bus)
+    isl_empty = set.intersection(isl_no_load, isl_no_gen)
+
+    if len(isl_empty) == 0:
+        return n
+
+    n.mremove("Bus", isl_empty)
+    n.determine_network_topology()
+
+    load_sum_final = n.loads_t.p_set.sum().sum()
+    generators_sum_final = n.generators.p_nom.sum()
+
+    logger.info(
+        f"Dropped {len(isl_empty)} buses. A resulted load discrepancy is {(100 * ((load_sum_final - load_sum_origin)/load_sum_origin)):2.1}% and {(100 * ((generators_sum_final - generators_sum_origin)/generators_sum_origin)):2.1}% for average load and generation capacity, respectivelly"
+    )
+
+    return n
+
+
+def drop_isolated_nodes(n, drop_e_threshold, n_periods_in_year=8760):
+    n.determine_network_topology()
+
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    loads_t_ann = n_periods_in_year * n.loads_t.p_set.mean(axis=0)
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+
+    load_islands = loads_t_ann.loc[loads_t_ann.index.isin(i_islands)]
+    # drop_e_threshold, GWh
+    small_load_index = load_islands[load_islands <= (1e3 * drop_e_threshold)].index
```

Comment:
I'd remove the 1e3 here, and move it below when this function is called.

Reply:
Let's try MW? Then we'd not need not 10e3 neither 8760

Reply:
Ok let's go for MW but we need to clarify average hourly demand.

Reply:
Revised